lib/src/controls/flyouts/menu.dart

```diff
@@ -0,0 +1,532 @@
+      const ClampingScrollPhysics();
+}
+
+/// See also:
+///
+///  * [MenuFlyout], which displays a list of commands or options
+///  * [MenuFlyoutItem], a single item in the list of items
+///  * [MenuFlyoutSeparator], which represents a horizontal line that
+///    separates items in a [MenuFlyout].
+///  * [MenuFlyoutSubItem], which represents a menu item that displays a
+///    sub-menu in a [MenuFlyout]
+///  * [MenuFlyoutItemBuilder], which renders the given widget in the items list
+abstract class MenuFlyoutItemBase {
+  final Key? key;
+
+  const MenuFlyoutItemBase({this.key});
+
+  Widget build(BuildContext context);
+}
+
+/// Render any widget in the items list of a [MenuFlyout]
+///
+/// See also:
+///
+///  * [MenuFlyout], which displays a list of commands or options
+///  * [MenuFlyoutItem], a single item in the list of items
+///  * [MenuFlyoutSeparator], which represents a horizontal line that
+///    separates items in a [MenuFlyout].
+///  * [MenuFlyoutSubItem], which represents a menu item that displays a
+///    sub-menu in a [MenuFlyout]
+class MenuFlyoutItemBuilder extends MenuFlyoutItemBase {
+  final WidgetBuilder builder;
+
+  /// Creates a menu flyout item builder
+  const MenuFlyoutItemBuilder({
+    super.key,
+    required this.builder,
+  });
+
+  @override
+  Widget build(BuildContext context) => builder(context);
+}
+
+/// The standart flyout item used inside a [MenuFlyout]
+///
+/// See also:
+///
+///  * [MenuFlyout], which displays a list of commands or options
+///  * [MenuFlyoutSeparator], which represents a horizontal line that
+///    separates items in a [MenuFlyout].
+///  * [MenuFlyoutSubItem], which represents a menu item that displays a
+///    sub-menu in a [MenuFlyout]
+class MenuFlyoutItem extends MenuFlyoutItemBase {
+  /// Creates a menu flyout item
+  MenuFlyoutItem({
+    super.key,
+    this.leading,
+    required this.text,
+    this.trailing,
+    required this.onPressed,
+    this.selected = false,
+  });
+
+  /// Displayed before [text].
+  ///
+  /// Don't feel obligated to provide icons for commands that don't have a
+  /// standard visualization. Cryptic icons aren't helpful, create visual
+  /// clutter, and prevent users from focusing on the important menu items.
+  ///
+  /// ![](https://learn.microsoft.com/en-us/windows/apps/design/controls/images/contextmenu_rs2_icons.png)
+  ///
+  /// Usually an [Icon] widget,
+  final Widget? leading;
+  final Widget text;
+  final Widget? trailing;
+  final VoidCallback? onPressed;
+  final bool selected;
+
+  bool _useIconPlaceholder = false;
+
+  @override
+  Widget build(BuildContext context) {
+    final size = ContentSizeInfo.of(context).size;
+    return Container(
+      width: size.isEmpty ? null : size.width,
+      padding: MenuFlyout.itemsPadding,
+      child: FlyoutListTile(
+        selected: selected,
+        showSelectedIndicator: false,
+        icon: leading ??
+            () {
```

Comment:
Avoid this kind of lambda calls for logic, either make a new function for this, compute this before returning or use a ternary operator. In this case i'd suggest removing this logic entirely and just returning null

Reply:
>  In this case i'd suggest removing this logic entirely and just returning null

`Icon(null)` makes the box have the same size as if it had an icon data. It's used if there is a placeholder